cachi2/core/package_managers/generic/main.py

```diff
@@ -102,3 +103,28 @@ def _load_lockfile(lockfile_path: RootedPath, output_dir: RootedPath) -> Generic
                 ),
             )
     return lockfile
+
+
+def _generate_sbom_components(lockfile: GenericLockfileV1) -> list[Component]:
+    """Generate a list of SBOM components for a given lockfile."""
+    components: list[Component] = []
+
+    for artifact in lockfile.artifacts:
+        name = Path(artifact.target).name
+        url = str(artifact.download_url)
+        component = Component(
+            name=name,
+            purl=PackageURL(
+                type="generic",
+                name=name,
+                qualifiers={
+                    "download_url": url,
+                    "checksums": ",".join([f"{a}:{h}" for a, h in artifact.checksums.items()]),
```

Comment:
Maybe one more suggestion, `a` and `h` do not tell much about what it stands for.  Btw. in the `LockfileArtifact` you have used: `ChecksumInfo(algo, digest) for algo, digest in self.checksums.items()`. For readability, you can define checksums just below the `url` variable.

But, I will approve the PR.

Reply:
I agree, it dows improve readability. Fixed.